### PR TITLE
Paging and page count for scaled out ServiceControl

### PR DESCRIPTION
**Pull request by adamralph**
Connects to #774

### CODE REVIEW

**Comment by adamralph** [comment]
@danielmarbach any objection to merging?

**Comment by WilliamBZA** [comment]
Shouldn't this be squashed before merging @adamralph?

**Comment by adamralph** [comment]
@WilliamBZA no, there are no fixups here. Each commit is self contained.

**Comment by adamralph** [comment]
I just noticed that ne of the commit messages has a weird typo:

> Refactor: Introduce MessageList/MessageListViewModel.RefreshMessages(string, bool)

It should be:

> Refactor: Introduce MessageListViewModel.RefreshMessages(string, bool)

I can fix that before we merge if preferred.

File: src/ServiceInsight/Search/SearchBarViewModel.cs [hunk 1]
```diff
@@ -52,22 +51,22 @@ protected override void OnActivate()
...
         public void GoToNextPage()
         {
-            Parent.RefreshMessages(SelectedEndpoint, CurrentPage + 1, SearchQuery);
+            Parent.RefreshMessages(NextLink);
         }
 
         public void GoToLastPage()
         {
-            Parent.RefreshMessages(SelectedEndpoint, PageCount, SearchQuery);
+            Parent.RefreshMessages(LastLink);
```
Comment (danielmarbach) [inline src/ServiceInsight/Search/SearchBarViewModel.cs]: I must say I preferred the previous name "NavigateTo" but that is just my taste

**Comment by danielmarbach** [comment]
@adamralph go ahead, I approved

File: src/ServiceInsight/Search/SearchBarViewModel.cs (hunk 1 above)
Comment (adamralph) [reply src/ServiceInsight/Search/SearchBarViewModel.cs]: That's also refactoring we could consider. It doesn't have to be part of this PR.

**Pull request merged**